azurerm/resource_arm_postgresql_configuration_test.go

```diff
@@ -0,0 +1,235 @@
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMPostgreSQLConfiguration_backslashQuote(t *testing.T) {
+	ri := acctest.RandInt()
+	location := testLocation()
+	config := testAccAzureRMPostgreSQLConfiguration_backslashQuote(ri, location)
+	serverOnlyConfig := testAccAzureRMPostgreSQLConfiguration_empty(ri, location)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPostgreSQLConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPostgreSQLConfigurationValue("azurerm_postgresql_configuration.test", "on"),
+				),
+			},
+			{
+				Config: serverOnlyConfig,
+				Check: resource.ComposeTestCheckFunc(
+					// "delete" resets back to the default value
+					testCheckAzureRMPostgreSQLConfigurationValueReset(ri, "backslash_quote"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPostgreSQLConfiguration_clientMinMessages(t *testing.T) {
+	ri := acctest.RandInt()
+	location := testLocation()
+	config := testAccAzureRMPostgreSQLConfiguration_clientMinMessages(ri, location)
+	serverOnlyConfig := testAccAzureRMPostgreSQLConfiguration_empty(ri, location)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPostgreSQLConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPostgreSQLConfigurationValue("azurerm_postgresql_configuration.test", "DEBUG5"),
+				),
+			},
+			{
+				Config: serverOnlyConfig,
+				Check: resource.ComposeTestCheckFunc(
+					// "delete" resets back to the default value
+					testCheckAzureRMPostgreSQLConfigurationValueReset(ri, "client_min_messages"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPostgreSQLConfiguration_deadlockTimeout(t *testing.T) {
+	ri := acctest.RandInt()
+	location := testLocation()
+	config := testAccAzureRMPostgreSQLConfiguration_deadlockTimeout(ri, location)
+	serverOnlyConfig := testAccAzureRMPostgreSQLConfiguration_empty(ri, location)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPostgreSQLConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPostgreSQLConfigurationValue("azurerm_postgresql_configuration.test", "5000"),
+				),
+			},
+			{
+				Config: serverOnlyConfig,
+				Check: resource.ComposeTestCheckFunc(
+					// "delete" resets back to the default value
+					testCheckAzureRMPostgreSQLConfigurationValueReset(ri, "deadlock_timeout"),
```

Comment:
Is there any particular reason you decided to implement your own check function instead of using the helper library?

Reply:
Because we're not checking the local state here, but the remote API (given there's no delete available and we're resetting this back to the original value)